x-pack/plugins/lens/public/editor_frame_service/embeddable/embeddable_factory.ts

```diff
@@ -20,9 +21,11 @@ import {
   IContainer,
 } from '../../../../../../src/plugins/embeddable/public';
 import { Embeddable } from './embeddable';
-import { SavedObjectIndexStore, DOC_TYPE } from '../../persistence';
+import { SavedObjectIndexStore, DOC_TYPE, getFilterableIndexPatternIds } from '../../persistence';
 import { getEditPath } from '../../../common';
 import { UiActionsStart } from '../../../../../../src/plugins/ui_actions/public';
+import { buildExpression } from '../editor_frame/expression_helpers';
```

Comment:
Thanks for reviewing so thoroughly on both code and architectural level. I realize I never wrote down the reasons to remove the expression, so this is definitely on me.

There are two general reasons why I think removing the expression completely for now is the right thing to do:
* As the state is redundant with the expression, it's always a source of potential errors in migrations - we basically need to re-implement a small part of the expression building logic in respective migrations and hope it matches what the actual expression building does. In fact, it's what often makes migrations necessary in the first place because changes are limited to the expression. By not saving the expression, this source of bugs (and code/complexity in general) can be removed. We originally decided to save the expression to avoid initializing the datasources in the embeddable, but this is a purely theoretical problem at the moment, as we only need access to index patterns which will be cached already anyway.
* Specifically, the app arch team plans to change the parameter structure of the `esaggs` function from a JSON string to nested sub expressions. When regenerating the expression, this change is trivial (just a single line in the `toExpression` function). With saved expressions, it becomes much more complex, because the logic for generating the nested sub expression ast is not available at migration time.

If your concern is the different code generating the expression in editor vs. embeddable, I could refactor the shared logic into a stateless helper that's used in both places in the same way. Do you think that would work?